### DISCUSSION TITLE
X86 Syscall Emulation on ARM-based MacBook

### DISCUSSION THREAD

**Discussion by pradyungn** [gem5/gem5#782] [Q&A] [answered]
Hi there! Currently attempting to do a syscall emulation sim (i.e se.py) with an X86 core, but running into the following issue:

```
src/base/loader/image_file_data.cc:112: fatal: fatal condition fd < 0 occurred: Failed to open file /lib64/ld-linux-x86-64.so.2.
This error typically occurs when the file path specified is incorrect.
```

I imagine that this is because I currently don't have any glibc headers installed... cuz I'm not on linux, or on x86. I was thinking of downloading compiled glibc headers manually, then forwarding those as the lib argument to gem5. Curious if anyone has tried something similar, or whether you all could suggest a different alternative? Thanks in advance :)

**Comment by pranith** [comment]
Try compiling a static binary or create a x86 based disk image with the dynamic binary. Either should work.

**Reply by pradyungn** [reply]
Sorry just wanted to clarify - would this be because of the gem5 binary itself lacking library files or because the specific program I am running is lacking library files?

I'm also a little confused on what exactly I should be statically compiling here.

**Reply by pradyungn** [reply]
I have tracked down the failure to the SEWorkload call in the se.py config file. Not too sure how to proceed from here.

**Reply by pranith** [reply, marked as answer]
Did you try compiling a static binary? When you execute a dynamic x86 binary, it looks for x86 architecture specific libraries. But since you are on ARM macbook, you cannot find those x86 libraries.

**Reply by pradyungn** [reply]
Yep, I ended up obtaining static binaries of the X86 program, and that made it work. Thanks for the suggestion!
